voucherify-service/src/commercetools/utils/getCommercetoolstCurrentPriceAmount.ts

```diff
@@ -1,6 +1,34 @@
 import { Product } from '@commercetools/platform-sdk';
 import { PriceSelector } from '../types';
 
+const filterByDistributionChannels = (priceVariant, distributionChannels) => {
+  if (!(distributionChannels?.length > 0) && !priceVariant?.channel) {
+    return true;
+  }
+  if (distributionChannels.length > 0 && !priceVariant?.channel) {
+    return false;
+  }
+  distributionChannels.forEach((currentChannel) => {
+    if (
+      priceVariant.channel.typeId === currentChannel.typeId &&
+      priceVariant.channel.id === currentChannel.id
+    ) {
+      return true;
```

Comment:
This condition does not work. It only terminates the function provided to forEach as a parameter.